plugins/scaffolder-node/src/actions/createTemplateAction.ts

```diff
@@ -150,16 +150,40 @@ export function createTemplateAction<
     TOutputSchema
   >,
 ): TemplateAction<TActionInput, TActionOutput, 'v2'> {
-  const { inputSchema, outputSchema } = parseSchemas(
+  const { inputSchema, outputSchema, inputZodSchema } = parseSchemas(
     action as TemplateActionOptions<any, any, any>,
   );
 
+  const wrapperHandler = (ctx: ActionContext<TActionInput, TActionOutput>) => {
```

Comment:
Hmm... The function isn't correct in isolation... It can't be that `ctx.input` and the final `schema.input` have the same type `TActionInput` 😅 This is apparent from the overuse of type assertions within this function.